nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -137,25 +151,26 @@ WaypointFollower::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
-void
-WaypointFollower::followWaypoints()
+template<typename T, typename U, typename V, typename Z>
+void WaypointFollower::followWaypointsLogic(
```

Comment:
This was honestly much better done than I could have hoped. This is great stuff. I expected to break out this one into several other more-primitive functions that they callbacks used, but if this works, power to it! Can we un-template though the waypoint type (e.g. PoseStamped)? They both use PoseStamped so it seems a little overkill to template that part.